crates/bevy_diagnostic/src/diagnostic.rs

```diff
@@ -172,8 +182,13 @@ impl Diagnostic {
     #[must_use]
     pub fn with_max_history_length(mut self, max_history_length: usize) -> Self {
```

Comment:
Isn't that just a constructor? I'm not sure I understand when this would be used where history isn't empty.

Reply:
To get a larger picture of a diagnostic over time. With `mem::swap` a user could get move access to a diagnostic from the store and change the max history length.

Reply:
Right, I guess that's a bit of a niche use case, but it doesn't hurt anything